package/linux-pam/0006-faillock-pipe.patch

```diff
@@ -0,0 +1,129 @@
+ /* If neither TALLY_FLAG_RHOST nor TALLY_FLAG_TTY are set the source is service. */
+ 
+ struct	tally {
+--- a/modules/pam_faillock/main.c
++++ b/modules/pam_faillock/main.c
+@@ -166,6 +166,12 @@
+ 		}
+ 
+ 		printf("%s:\n", user);
++		// Siklu addition
++		uint16_t locked = 0;
++		if (tallies.count > 0)
++			locked = tallies.records[0].status & TALLY_STATUS_LOCKED;
++		printf("%s\n", locked ? "Locked" : "Not locked");
++		// end of Siklu addition
+ 		printf("%-19s %-5s %-48s %-5s\n", "When", "Type", "Source", "Valid");
+ 
+ 		for (i = 0; i < tallies.count; i++) {
+--- a/modules/pam_faillock/pam_faillock.c
++++ b/modules/pam_faillock/pam_faillock.c
+@@ -46,6 +46,10 @@
+ #include <syslog.h>
+ #include <ctype.h>
+ 
++#include <sys/types.h>
++#include <sys/stat.h>
++#include <fcntl.h>
++
+ #ifdef HAVE_LIBAUDIT
+ #include <libaudit.h>
+ #endif
+@@ -71,6 +75,10 @@
+ 
+ #define MAX_TIME_INTERVAL 604800 /* 7 days */
+ #define FAILLOCK_CONF_MAX_LINELEN 1023
++// Siklu addition
++#define PAM_FIFO                   "/tmp/pam-fifo"
++#define PAM_FIFO_MSG_MAXSIZE       127
++// end of Siklu addition
+ 
+ static const char default_faillock_conf[] = FAILLOCK_DEFAULT_CONF;
+ 
+@@ -452,6 +460,11 @@
+ 		return PAM_SUCCESS;
+ 	}
+ 
++	// Siklu addition
++	if (tallies->count > 0)
++		tallies->records[0].status &= ~TALLY_STATUS_LOCKED;
++	// end of Siklu addition
++
+ 	latest_time = 0;
+ 	for (i = 0; i < tallies->count; i++) {
+ 		if ((tallies->records[i].status & TALLY_STATUS_VALID) &&
+@@ -624,6 +637,31 @@
+ 	return PAM_SYSTEM_ERR;
+ }
+ 
++// Siklu addition
++static int
++write_tally_lock(pam_handle_t *pamh, struct options *opts, struct tally_data *tallies, int *fd)
++{
++	if (*fd == -1) {
++		*fd = open_tally(opts->dir, opts->user, opts->uid, 1);
++	}
++	if (*fd == -1) {
++		if (errno == EACCES) {
++			return PAM_SUCCESS;
++		}
++		pam_syslog(pamh, LOG_ERR, "Error opening the tally file for %s: %m", opts->user);
++		return PAM_SYSTEM_ERR;
++	}
++
++	if (tallies->count > 0)
++		tallies->records[0].status |= TALLY_STATUS_LOCKED;
++
++	if (update_tally(*fd, tallies) == 0)
++		return PAM_SUCCESS;
++
++	return PAM_SYSTEM_ERR;
++}
++// end of Siklu addition
++
+ static void
+ faillock_message(pam_handle_t *pamh, struct options *opts)
+ {
+@@ -644,6 +682,25 @@
+ 
+ 			pam_info(pamh, _("(%d minutes left to unlock)"), (int)left);
+ 		}
++
++		// Siklu addition
++		int fd = open(PAM_FIFO, O_WRONLY, 0600);
++		if (fd == -1) {
++			pam_syslog(pamh, LOG_ERR, "Error opening fifo %s", PAM_FIFO);
++			return;
++		}
++
++		char buf[PAM_FIFO_MSG_MAXSIZE + 1];
++		buf[PAM_FIFO_MSG_MAXSIZE] = 0;
++		snprintf(buf, PAM_FIFO_MSG_MAXSIZE, "The account of user %s is locked due to %u failed logins.",
++			opts->user, (unsigned int)opts->failures);
++
++		int len = strlen(buf);
++		int res = write(fd, buf, len);
++		close(fd);
++		if (res != len) 
++			pam_syslog(pamh, LOG_ERR, "Error writing to fifo %s: expected %d, actual %d", PAM_FIFO, len, res);
++		// end of Siklu addition
```

Comment:
i think these are the only lines that required
all other are useless and should be removed from patch

and again -- c++ comments -- no-go

Reply:
and did you put it in proper scope ?
i think it should be just after `pam_info` call, not outside of scope

now you have mismatch between ssh terminal output and user-log records